src/llmq/options.cpp

```diff
@@ -116,26 +116,26 @@ std::map<Consensus::LLMQType, QvvecSyncMode> GetEnabledQuorumVvecSyncEntries()
 
 bool IsQuorumTypeEnabled(Consensus::LLMQType llmqType, gsl::not_null<const CBlockIndex*> pindexPrev)
 {
-    return IsQuorumTypeEnabledInternal(llmqType, pindexPrev, std::nullopt, std::nullopt);
+    return IsQuorumTypeEnabledInternal(llmqType, pindexPrev, std::nullopt);
 }
 
 bool IsQuorumTypeEnabledInternal(Consensus::LLMQType llmqType, gsl::not_null<const CBlockIndex*> pindexPrev,
-                                std::optional<bool> optDIP0024IsActive, std::optional<bool> optHaveDIP0024Quorums)
+                                 std::optional<bool> optIsDIP0024Active)
 {
     const Consensus::Params& consensusParams = Params().GetConsensus();
 
-    const bool fDIP0024IsActive{optDIP0024IsActive.value_or(DeploymentActiveAfter(pindexPrev, consensusParams, Consensus::DEPLOYMENT_DIP0024))};
-    const bool fHaveDIP0024Quorums{optHaveDIP0024Quorums.value_or(pindexPrev->nHeight >= consensusParams.DIP0024QuorumsHeight)};
+    const bool fDIP0024IsActive{
+        optIsDIP0024Active.value_or(DeploymentActiveAfter(pindexPrev, consensusParams, Consensus::DEPLOYMENT_DIP0024))};
```

Comment:
nit: we should rework this, currently this will always call DeploymentActiveAfter as it's a param. Should put in a lambda or something to avoid calling DeploymentActiveAfter if optIsDIP0024Active is set

Reply:
we discussed it, this `optIsDIP0024Active` is used only for Unit test and it means that on production (mainnet, testnet, any devnet) it would be a case of `value_or`. So, this optimization won't give any extra benefits.

Also just a side note (not really important in this particular case) the function `DeploymentActiveAfter` for buried deployment (which is DIP0024) is simplified to one single `if` condition with height comparision, so, there's not much calculation is done anyway